integrations/airflow/marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -115,16 +115,29 @@ def extract_on_complete(self, task_instance) -> [StepMetadata]:
             context=context
         )]
 
+    def get_parsed_table_inputs(self,data):
+        data1 = re.sub(r'[?|`]',r'',data)
+        datas = re.split('[.]',str(data1))
+        table_dict = {"projectId":datas[0],"datasetId":datas[1],"tableId":datas[2]}
+        return table_dict
+
     def _get_input_from_bq(self, job, context, source, client):
-        if not job._properties.get('statistics')\
+        
+        if job._properties.get('statistics').get('query').get('referencedTables') is not None:
+            bq_input_tables = job._properties.get('statistics')\
+            .get('query')\
+            .get('referencedTables')
+
+        elif context['bigquery.sql.parsed.inputs'] is not None:
```

Comment:
The main usage of the context key `bigquery.sql.parsed.inputs` is to compare the input and output tables returned via the `BigQuery` client call. For `BigQueryOperator`, we use the job details object (see example [here](https://github.com/MarquezProject/marquez/blob/main/integrations/airflow/tests/extractors/job_details.json)) returned by the client to set the in / out tables, so we wouldn't want to default to using the context here (though that might be an option in the future).

Note that the context key also helps to improve the [SQL parser](https://github.com/MarquezProject/marquez/blob/main/integrations/airflow/marquez_airflow/extractors/sql/experimental/parser.py) shipped with the `marquez-airflow` lib if they differ from wants returned by `BigQuery`